src/vmm/src/lib.rs

```diff
@@ -493,6 +498,116 @@ impl Vmm {
             .set_kvm_memory_regions(&self.guest_memory, enable)
             .map_err(Error::Vm)
     }
+
+    /// Updates configuration for the balloon device as described in `balloon_update`.
+    fn latest_balloon_stats(&self) -> std::result::Result<VmmData, VmmActionError> {
+        if let Some(busdev) = self.get_bus_device(DeviceType::Virtio(TYPE_BALLOON), BALLOON_DEV_ID)
+        {
+            let virtio_device = busdev
+                .lock()
+                .expect("Poisoned lock")
+                .as_any()
+                .downcast_ref::<MmioTransport>()
+                // Only MmioTransport implements BusDevice at this point.
+                .expect("Unexpected BusDevice type")
+                .device();
+
+            let latest_stats = virtio_device
+                .lock()
+                .expect("Poisoned lock")
+                .as_mut_any()
+                .downcast_mut::<Balloon>()
+                .unwrap()
+                .latest_stats()
+                .map(|stats| VmmData::BalloonStats(stats.clone()));
+
+            latest_stats.ok_or(VmmActionError::BalloonConfig(
+                BalloonConfigError::StatsNotFound,
+            ))
+        } else {
+            Err(VmmActionError::BalloonConfig(
+                BalloonConfigError::DeviceNotFound,
+            ))
+        }
+    }
+
+    /// Updates configuration for the balloon device as described in `balloon_update`.
+    fn update_balloon_config(&mut self, balloon_update: BalloonUpdateConfig) -> ActionResult {
```

Comment:
This can directly take `amount_mb`.

Either that, or define `BalloonUpdateConfig` as part of the `devices::virtio::balloon` if it makes sense.